scripts/website/generate.go

```diff
@@ -0,0 +1,206 @@
+// ToDo (nilekh): Get this struct from the Gatekeeper repo
+type ConstraintTemplate struct {
+	APIVersion string `yaml:"apiVersion"`
+	Kind       string `yaml:"kind"`
+	Metadata   struct {
+		Name        string `yaml:"name"`
+		Annotations struct {
+			Title       string `yaml:"metadata.gatekeeper.sh/title"`
+			Description string `yaml:"description"`
+		} `yaml:"annotations"`
+	} `yaml:"metadata"`
+	Spec struct {
+		Crd struct {
+			Spec struct {
+				Names struct {
+					Kind string `yaml:"kind"`
+				} `yaml:"names"`
+				Validation struct {
+					OpenAPIV3Schema struct {
+						Type       string `yaml:"type"`
+						Properties struct {
+							Repos struct {
+								Description string `yaml:"description"`
+								Type        string `yaml:"type"`
+								Items       struct {
+									Type string `yaml:"type"`
+								} `yaml:"items"`
+							} `yaml:"repos"`
+						} `yaml:"properties"`
+					} `yaml:"openAPIV3Schema"`
+				} `yaml:"validation"`
+			} `yaml:"spec"`
+		} `yaml:"crd"`
+		Targets []struct {
+			Target string `yaml:"target"`
+			Rego   string `yaml:"rego"`
+		} `yaml:"targets"`
+	} `yaml:"spec"`
+}
+
+func main() {
+	pwd, err := os.Getwd()
+	if err != nil {
+		fmt.Println("error while getting pwd")
+		panic(err)
+	}
+
+	rootDir := filepath.Join(pwd, "..", "..")
+	libraryPath := filepath.Join(rootDir, "library")
+	dirEntry, err := os.ReadDir(libraryPath)
+	if err != nil {
+		fmt.Println("error while listing directories under library")
+		panic(err)
+	}
+
+	for _, entry := range dirEntry {
+		if entry.Type().IsDir() {
+			basePath, _ := filepath.Abs(filepath.Join(libraryPath, entry.Name()))
+			directories, err := os.ReadDir(basePath)
+			if err != nil {
+				fmt.Println("error while listing directories under ", entry.Name())
+				panic(err)
+			}
+
+			for _, dir := range directories {
+				if dir.Type().IsDir() {
+					fmt.Println("Generating markdown for ", filepath.Join(basePath, dir.Name()))
+
+					suiteContent, err := os.ReadFile(filepath.Join(basePath, dir.Name(), "suite.yaml"))
+					if err != nil {
+						fmt.Println("error while reading suite.yaml")
+						panic(err)
+					}
+
+					suite := Suite{}
+					yaml.Unmarshal(suiteContent, &suite)
+
+					// ConstraintTemplate
+					constraintTemplateContent, err := os.ReadFile(filepath.Join(basePath, dir.Name(), "template.yaml"))
+					if err != nil {
+						fmt.Println("error while reading template.yaml")
+						panic(err)
+					}
+					constraintTemplate := ConstraintTemplate{}
+					yaml.Unmarshal(constraintTemplateContent, &constraintTemplate)
+
+					allExamples := ""
+					for _, test := range suite.Tests {
+						constraintContent, err := os.ReadFile(filepath.Join(basePath, dir.Name(), test.Constraint))
+						if err != nil {
+							fmt.Println("error while reading constraint.yaml")
+							panic(err)
+						}
+						constraintExample := fmt.Sprintf("<details>\n<summary>constraint</summary>\n\n```yaml\n%s\n```\n\n</details>\n", constraintContent)
```

Comment:
If you wanted a templating-style solution, you could use go templates:

https://pkg.go.dev/text/template